django_netjsongraph/tests/base/test_admin.py

```diff
@@ -95,6 +96,32 @@ def test_topology_receive_url(self):
         self.assertEqual(response.status_code, 200)
         self.assertContains(response, 'field-receive_url')
 
+    def test_custom_topology_receive_url(self):
+        t = self.topology_model.objects.first()
+        t.strategy = 'receive'
+        t.save()
+        path = reverse('{0}_topology_change'.format(self.prefix), args=[t.pk])
+        # No change in URL Test
+        response = self.client.get(path)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, 'field-receive_url')
+        self.assertContains(response, 'http://testserver/api/receive')
+        # Change URL Test
+        AbstractTopologyAdmin.receive_url_baseurl = 'http://changedurlbase'
+        response = self.client.get(path)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, 'field-receive_url')
+        self.assertContains(response, 'http://changedurlbase/api/receive')
+        # Change URLConf Test
+        AbstractTopologyAdmin.receive_url_urlconf = 'django_netjsongraph.api.urls'
```

Comment:
we can't hardcode `django_netjsongraph.api.urls` here otherwise tests in openwisp-network-topology fail because they'll try to load `django_netjsongraph` which won't be in `INSTALLED_APPS`.

Try the following:

```python
#put this at the top
module = 'django_netjsongraph'

# put this in the test
AbstractTopologyAdmin.receive_url_urlconf = '{}.api.urls'.format(self.module)
```

Reply:
Oh! Now it makes sense, I was having some errors and was looking into it, thanks! :smile:

Reply:
Yep, it worked, thanks! :smile: